extras/axelar-token/src/IpcTokenHandler.sol

```diff
@@ -0,0 +1,115 @@
+    //         the call originated at the Axelar ITS.
+    function _executeWithInterchainToken(
+        bytes32, // commandId
+        string calldata, // sourceChain
+        bytes calldata, // sourceAddress
+        bytes calldata data,
+        bytes32, // tokenId
+        address tokenAddr,
+        uint256 amount
+    ) internal override {
+        (SubnetID memory subnet, address recipient) = abi.decode(data, (SubnetID, address));
+
+        IERC20 token = IERC20(tokenAddr);
+        require(token.balanceOf(address(this)) >= amount, "insufficient balance");
+
+        // Authorize the IPC gateway to spend these tokens on our behalf.
+        token.approve(address(_ipcGateway), amount);
+
+        // Fund the designated subnet via the IPC gateway.
+        _ipcGateway.fundWithToken(subnet, recipient.from(), amount);
+
+        // Emit an event.
+        emit SubnetFunded(subnet, recipient, amount);
+    }
+
+    // @notice Handles result messages for funding operations.
+    function handleIpcMessage(IpcEnvelope calldata envelope) external payable returns (bytes memory ret) {
+        if (msg.sender != address(_ipcGateway)) {
+            revert IpcHandler.CallerIsNotGateway();
+        }
+        if (envelope.kind != IpcMsgKind.Result) {
+            revert IpcHandler.UnsupportedMsgKind();
+        }
+
+        ResultMsg memory result = abi.decode(envelope.message, (ResultMsg));
+        if (result.outcome != OutcomeType.Ok) {
+            // Note: IPC only supports deploying subnets via our blessed registry, so we can trust the code behind
+            // the subnet actor.
+            SupplySource memory supplySource = SubnetActor(envelope.from.subnetId.getAddress()).supplySource();
+            require(supplySource.kind == SupplyKind.ERC20, "expected ERC20 supply source");
+
+            // Results will carry the original beneficiary in the 'from' address.
+            address beneficiary = envelope.from.rawAddress.extractEvmAddress();
+
+            // We credit the token funds to the beneficiary. The beneficiary will have to call withdraw() to pull the
+            // funds out on this network.
+            _claims[beneficiary][supplySource.tokenAddress] += envelope.value;
+
+            // Emit an event.
+            emit FundingFailed(envelope.from.subnetId, beneficiary, envelope.value);
+        }
+
+        return bytes("");
+    }
+
+    // @notice Withdraws all available balance for the specified token for the sender.
+    function withdraw(address token) external {
+        uint256 available = _claims[msg.sender][token];
+        if (available == 0) {
+            revert NothingToWithdraw();
+        }
+
+        delete _claims[msg.sender][token];
+        IERC20(token).safeTransfer(msg.sender, available);
+    }
+
+    // @notice Queries the claim of a beneficiary over a particular token.
+    function claimFor(address beneficiary, address token) external view returns (uint256) {
```

Comment:
claimFor as a name implies a change is there a name like getClaimFor that we can use to make it more clearly a view?

Reply:
Good point! The convention for getters is to not prefix them with `get`, but agree that it becomes ambiguous in this case as `claim` can be confused for a verb/action. Will change.